blogposts/2021/optimizing-a-code-intelligence-commit-graph-2.md

```diff
@@ -302,19 +304,19 @@ This made things **far** worse. In Sourcegraph 3.22, we moved the code that calc
 
 This made their entire instance unstable, and this series of events escalated us from a priority zero event to a [priority now](https://devblogs.microsoft.com/oldnewthing/20081121-00/?p=20123) event.
 
-We attacked the problem again, starting out again by tackling some low-hanging fruit by ignoring large amounts of unneeded data. When we pull back the commit graph for a repository, it's unlikely that we need the _entire_ commit graph. There's little sense in filling out the visibility of the long tale of historic commits, especially as its distance to the oldest LSIF upload grows over time. Now, we [entirely ignore](https://github.com/sourcegraph/sourcegraph/pull/16140) the portion of the commit graph that existed _before_ the oldest known LSIF upload for that repository.
+We attacked the problem again, starting out once more by tackling some low-hanging fruit, by ignoring large amounts of unneeded data. When we pull back the commit graph for a repository, it's unlikely that we need the _entire_ commit graph. There's little sense in filling out the visibility of the long tale of historic commits, especially as its distance to the oldest LSIF upload grows over time. Now, we [entirely ignore](https://github.com/sourcegraph/sourcegraph/pull/16140) the portion of the commit graph that existed _before_ the oldest known LSIF upload for that repository.
```

Comment:
@efritz I just remembered a query I had on this – should it be "long tail" instead of "long tale"?

Reply:
Yes, should be _long tail_. I'll edit and resolve this comment once pushed.